src/doc/rustdoc/src/passes.md

```diff
@@ -1,3 +1,143 @@
 # Passes
 
-Coming soon!
\ No newline at end of file
+Rustdoc has a concept called "passes". These are transformations that
+`rustdoc` runs on your documentation before producing its final output.
+
+In addition to the passes below, check out the docs for these flags:
+
+* [`--passes`](command-line-arguments.html#--passes-add-more-rustdoc-passes)
+* [`--no-defaults`](ommand-line-arguments.html#--no-defaults-dont-run-default-passes)
+
+## Default passes
+
+By default, rustdoc will run some passes, namely:
+
+* `strip-hidden`
+* `strip-private`
+* `collapse-docs`
+* `unindent-comments`
+
+However, `strip-private` implies `strip-private-imports`, and so effectively,
+all passes are run by default.
+
+## `strip-hidden`
+
+This pass implements the `#[doc(hidden)]` attribute. When this pass runs, it
+checks each item, and if it is annotated with this attribute, it removes it
+from `rustdoc`'s output.
+
+Without this pass, these items will remain in the output.
+
+## `unindent-comments`
+
+When you write a doc comment like this:
+
+```rust,ignore
+/// This is a documentation comment.
+```
+
+There's a space between the `///` and that `T`. That spacing isn't intended
+to be a part of the output; it's there for humans, to help separate the doc
+comment syntax from the text of the comment. This pass is what removes that
+space.
+
+It takes this indentation amount from the first line. For example:
```

Comment:
What it seems to do is take the smallest indent from the whole document but it will ignore the indent on the first line if the second line isn't entirely white-space. It also always removes white-space from the beginning of the first line and doesn't remove anything from lines which are entirely white-space. I'm sure the logic can be improved because there are also #38173 and #38739 which I think are also caused by this.